src/future/mod.rs

```diff
@@ -145,6 +145,13 @@ where
     fn poll(mut self: Pin<&mut Self>, cx: &mut Context<'_>) -> Poll<Self::Output> {
         match self.future.as_mut().poll(cx) {
             Poll::Ready(result) => {
+                // If we've finished execution already (this task was detached), don't clean up. We
+                // can't access the state any more to destroy thread locals, and don't want to run
+                // any more wakers (which will be no-ops anyway).
+                if ExecutionState::try_with(|state| state.is_finished()).unwrap_or(true) {
+                    return Poll::Ready(());
+                }
```

Comment:
Why are we still polling the future if execution already finished?

Was the task's local storage already dropped in `ExecutionState::cleanup` when we drained tasks?

Reply:
We're not re-polling the future after execution finishes. But at `ExecutionState::cleanup` time, we drop the continuation, which involves _resuming_ the task so that it can unwind its stack. That's how we end up in this code. We do still clean up the task's local storage in`ExecutionState::cleanup`, so this change isn't actually leaking anything, just changing where the cleanup happens.